django/db/models/sql/compiler.py

```diff
@@ -2019,6 +2019,11 @@ def as_sql(self):
                         "Window expressions are not allowed in this query "
                         "(%s=%r)." % (field.name, val)
                     )
+                if isinstance(val, ColPairs):
+                    raise FieldError(
```

Comment:
We don't have a good hygiene of `ValueError` vs `FieldError` but the latter seemed most consistent with the other cases.